_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,363 @@
+
+## Background
+
+- [Understanding Success Criterion 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
+- [G18: Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G18)
+- [G145: Ensuring that a contrast ratio of at least 3:1 exists between text (and images of text) and background behind the text](https://www.w3.org/WAI/WCAG21/Techniques/general/G145)
+- [F83: Failure of Success Criterion 1.4.3 and 1.4.6 due to using background images that do not provide sufficient contrast with foreground text (or images of text)](https://www.w3.org/WAI/WCAG21/Techniques/failures/F83)
+- [CSS Scoping Module Level 1 (Editor's Draft)](https://drafts.csswg.org/css-scoping/)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This dark grey text is on a white background.
+
+```html
+<p style="color: #333; background: #FFF;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 2
+
+This dark grey text is on a partially white gradient background.
+
+```html
+<p style="color: #333; background: linear-gradient(to right, #FFF, #00F); width: 300px;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 3
+
+This light grey text is on a partially black background image.
+
+```html
+<p
+	style="color: #CCC; height:50px; padding-top:25px; background: #000 no-repeat -20px -20px url('../test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Passed Example 4
+
+This text passes because the text shadow gives the text sufficient contrast.
+
+```html
+<p style="color: #000; background: #737373; text-shadow: white 0 0 3px">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 5
+
+This text only has a 3:1 contrast but it is a large (18 point) text so it still passes.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in a human language
+</p>
+```
+
+#### Passed Example 6
+
+This text only has a 3:1 contrast but it is a large (14 points and bold) text so it still passes.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Passed Example 7
+
+This text has an insufficient contrast but it is decorative so it still passes.
+
+```html
+<p>Helvetica is a widely used sans-serif typeface developed in 1957 by Max Miedinger and Eduard Hoffmann.</p>
+<p style="font-family: helvetica; background: #EEE; color: #777;" aria-hidden="true">
+	The quick brown fox jumps over the lazy dog.
+</p>
+```
+
+#### Passed Example 8
+
+This text has an insufficient contrast but it does not express anything in human language so it still passes.
```

Comment:
This week-end I stumbled upon an interesting case for the "no human language" exception…
Consider this page: https://genius.com/Ishtar-band-o-julissi-lyrics
It's main content is the lyrics of the song _O julissi_ which "is sung entirely in Ukrainian-influenced imaginary language" [Wikipedia](https://en.wikipedia.org/wiki/O_Julissi)

Now, this does not express anything in human language, and was meant so. But I feel that the lyrics should be considered as text and have high contrast because they need to be readable…

Would you consider that:
- this is text content and requires 4.5:1 contrast?
- this is non-text content and requires 3:1 contrast?
- this does not express anything in human language and has no contrast requirement?

Reply:
I don't think this is relevant to the rule.

Reply:
So this is where Eurovision and WCAG collide :) The same case could be made for these lyrics: https://genius.com/Urban-trad-sanomi-eurovision-edit-lyrics (A far superior song. Also Belgian, make of that what you will).

Either way, I agree with @WilcoFiers here. I would not count these lyrics as written in a human language and this is where WCAG draws the line.